ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -102,11 +102,16 @@ public void run() {
     ExecutorService executor = getTimeoutHandlingExecutor();
     try {
       do {
-        Future<Boolean> singleRun = executor.submit(() -> findNextCompactionAndExecute(computeStats));
+        Ref<CompactionInfo> ci = new Ref<>(null);
+        Future<Boolean> singleRun = executor.submit(() -> findNextCompactionAndExecute(computeStats, ci));
```

Comment:
Wouldn't it be more natural to have a complex return type, like below? Definitely not a final proposal just some food for thought.
```
private static class ExecutionResult {
   private Boolean singleRun;
   private CompactionInfo failedCompaction;
}
```